fix: Fix "Standardize phone number identifiers" migration [CHI-2673]

GPaoloni opened the pull request:
## Description
This PR attempts to fix the (broken) migration merged in https://github.com/techmatters/hrm/pull/613.
The migration, initially, didn't contemplated "conflicts" on identifiers - that is, if `1234` and `12 34` are both identifiers stored in the DB, the migration fails to run.

This screenshots intend to show what are the effects of the whole migration
- Before migration, see the "conflicting identifiers"
  ![Screenshot from 2024-04-09 17-24-39](https://github.com/techmatters/hrm/assets/15805319/062a6d51-ce7a-4242-8d67-7564efd35afe)
- See which contacts are related to the above identifiers
  ![Screenshot from 2024-04-09 17-25-48](https://github.com/techmatters/hrm/assets/15805319/5c3d214c-1fa9-4d11-b7a0-378ae9c484c7)
- Run the queries, one by one (note how many records are updated/deleted on each step)
  ![Screenshot from 2024-04-09 17-27-34](https://github.com/techmatters/hrm/assets/15805319/b1d9943e-d520-473d-8b4e-179d307934fa)
  ![Screenshot from 2024-04-09 17-29-14](https://github.com/techmatters/hrm/assets/15805319/1f362c6b-eb87-4daa-bed1-9fefb30c3410)
  ![Screenshot from 2024-04-09 17-29-28](https://github.com/techmatters/hrm/assets/15805319/3d039c9d-61f4-46a4-87c8-4f5f5b18a145)
  ![Screenshot from 2024-04-09 17-30-19](https://github.com/techmatters/hrm/assets/15805319/c3d33a75-1c49-4ecd-a276-022ae5a349a6)

After all this queries, there are no identifiers with blank spaces in my local DB, neither any "conflicting identifiers".

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2673)
- [ ] New tests added
- [ ] Feature flags / configuration added

### Verification steps
<!--
Describe how to validate your changes.
- Include screen shots if applicable.
- Note if migrations are required.
-->

### AFTER YOU MERGE

1. Cut a release tag using the GitHub workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P

GPaoloni commented on hrm-domain/hrm-service/migrations/20240329215610-sanitize-identifiers-remove-blank-spaces.js:
```diff
@@ -19,17 +19,65 @@
+    `);
+    console.log('Conflicting identifiers deleted');
+
+    await queryInterface.sequelize.query(`
+      UPDATE "Identifiers" "idx" SET "identifier" = replace("idx"."identifier", ' ', '')
+      FROM (
+        SELECT DISTINCT identifiers.* FROM "Identifiers" identifiers
+              JOIN "Contacts" contacts ON identifiers.id = contacts."identifierId"
+              WHERE identifiers.identifier LIKE '% %' AND contacts."channel" IN ('voice', 'whatsapp', 'sms', 'modica')
+      ) AS "whitespaced" WHERE "idx"."id" = "whitespaced"."id"
+    `);
+    console.log('Whitespaced identifiers sanitized');
```
@stephenhand Note that here, because `AND contacts."channel" IN ('voice', 'whatsapp', 'sms', 'modica')`, blank-spaced identifiers that already exist in the DBs will not be reachable. I am not sure if this `WHERE` chunk should be present, as I don't think we can, but if we want to keep it we also need to tweek the "search contacts by identifier" endpoint to include the channel type of the task (since we'll want to remove white-spaces only from the above channels).

What do you think? Better to remove this and force non-white-spaced identifiers?

stephenhand reviewed (approved):
Wow, that turned out more complicated than we initially thought, didn't it? Thanks Gian

GPaoloni commented:
@stephenhand I've added https://github.com/techmatters/flex-plugins/pull/2247 to Flex, to handle the "sanitize identifier" part (and reverted that in hrm in [40a824e](https://github.com/techmatters/hrm/pull/620/commits/40a824ef1685d22c6e28f52a7c3db9075d7e6f7e)). Lemme know if this is aligned with your thoughts.
